### DISCUSSION TITLE
Different Table Name For Same Entity

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [SeaQL/sea-orm#1574] [Ideas]
## Motivation

Supplementary to https://github.com/surrealdb/surrealdb/pull/1717, I noticed that if we can map the semantic of SurrealDB for different tables, and this could be helpful if you have some database access that already have similar structures so we can just map for examples. "SurrealDB namespace" into database schema and "SurrealDB database" into database tables) 

Right now we need to have the table exactly named "kvstore" and it is pretty much not so cool because existing users that have similar table but named other than "kvstore" (maybe they like to call the table "surreal_datastore") cannot enjoy the backend.

## Proposed Solutions

Let us have the table name dynamically.

## Current Workarounds

Switching to SeaQuery wouldn't help as we needed transaction support but those are semantic based on the SQL driver...which means implementation-defined, and that you must use an ORM with it? 

Or I can just go one level lower, rewrite that thing from SeaORM into a combination of SQLx + SeaQuery which is tedious and maybe overlapping with SeaORM for 90% of the features, totally not worth it...

Another hacky way to do this is to use a special table name, capture the query generated by SeaORM, replace the special table name with our target table name, and that is going to be very hacky.

**Comment by tyt2y3** [comment]
Supporting SurrealDB is not on our roadmap, but regarding dynamic table name...

`EntityName` and `IdenStatic` is object safe, that means it can be dynamic. I imagine one can have a custom Entity like the following:

```rust
// as long as it impl Copy
pub struct Entity {
    i: usize,
}
```

and impl all other traits.

If you have time to do some experiments, I would like to hear your results. There might be blockers in other parts of the library.
